packages/loader/container-loader/src/container.ts

```diff
@@ -1149,8 +1160,9 @@ export class Container
 				this.mc.logger.sendTelemetryEvent(
 					{
 						eventName: "ContainerDispose",
-						// Only log error if container isn't closed
-						category: !this.closed && error !== undefined ? "error" : "generic",
+						// Use error category if there's an error, unless we already closed with an error (i.e. _criticalError is set)
+						category:
+							error !== undefined && this._criticalError === undefined ? "error" : "generic",
```

Comment:
why change this?

Reply:
I'm thinking that the `closed` state is a legit (non-error) state for a container to be in, if it was closed with no error (consider the fact that we wouldn't log that "ContainerClosed" event as an error).  Then if it's disposed _with_ an error, that'd be the point where we'd want to see the error log.

i.e. if you close (no error) then dispose (with error), before you would get no error logs.  Now you do get an error log.